osl/namespace_linux.go

```diff
@@ -157,12 +157,8 @@ func (n *networkNamespace) NeighborOptions() NeighborOptionSetter {
 	return n
 }
 
-func reexecCreateNamespace() {
-	if len(os.Args) < 2 {
-		log.Fatal("no namespace path provided")
-	}
-
-	if err := syscall.Mount("/proc/self/ns/net", os.Args[1], "bind", syscall.MS_BIND, ""); err != nil {
+func mountNetworkNamespace(basePath string, lnPath string) {
+	if err := syscall.Mount(basePath, lnPath, "bind", syscall.MS_BIND, ""); err != nil {
 		log.Fatal(err)
```

Comment:
It may not ok to do `log.Fatal` when you are in daemon context. It's ok to do that when we are in reexec context. So I would suggest that `mointNetworkNamespace` returns an error. While in the reexec context we can invoke `log.Fatal` and while daemon context we can simply return an error from `GetSandboxForExternalKey`